### PR TITLE
rationalize some whitespace & punctuation

### PR DESCRIPTION
**Pull request by TallTed**
***
<a href="https://pr-preview.s3.amazonaws.com/TallTed/sparql-update/pull/19.html" title="Last updated on Jun 30, 2023, 4:44 PM UTC (be751a4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/sparql-update/19/e788001...TallTed:be751a4.html" title="Last updated on Jun 30, 2023, 4:44 PM UTC (be751a4)">Diff</a>

### CODE REVIEW

**Review by rubensworks — APPROVED** [review]

File: spec/index.html [hunk 1]
```diff
@@ -1363,8 +1363,8 @@ <h4>Dataset-UNION</h4>
             <p>where union between graphs is defined as set-union of triples in those graphs.</p>
           </div>
           <div class="note">
-            <p class="prefix"><strong>Note:</strong></p>Note that, in the following, whenever we write Dataset-UNION( <em>X</em> ) where <em>X</em> = {DS<sub>1</sub>,DS<sub>2</sub>,...
-            ,DS<sub>n</sub>} is a set of datasets, we understand this as a shorthand for Dataset-UNION(DS<sub>1</sub>, Dataset-UNION(DS<sub>2</sub>, ... , Dataset-UNION(DS<sub>n</sub>,{})...)).
+            <p class="prefix"><strong>Note:</strong></p>Note that, in the following, whenever we write Dataset-UNION ( <em>X</em> ) where <em>X</em> = { DS<sub>1</sub>, DS<sub>2</sub>, ...,
+            DS<sub>n</sub> } is a set of datasets, we understand this as a shorthand for Dataset-UNION ( DS<sub>1</sub>, Dataset-UNION ( DS<sub>2</sub>, ..., Dataset-UNION ( DS<sub>n</sub>, {} ) ... ) ).
```
Comment (afs) [inline spec/index.html:1367]: This shows why the spaces are bad here. The nested forms do not look nested. Proper layout and styling is needed especially for narrow screed display.

**Comment by afs** [comment]
1/ Ted knows I do not support changing dashes the mdashes. The SPARQL docs systematically use dashes.

2/ White space around parentheses make no effort to find a middle ground on styling. 
The appearance becomes inconsistent with the style of the rest of the SPARQL documents and that is a lot of places.

**Comment by TallTed** [comment]
> 1/ Ted knows I do not support changing dashes the mdashes. The SPARQL docs systematically use dashes.

Ted forgets such things. Sorry. He participates in a number of WGs and CGs, and contributes to dozens of documents, and m-dashes are not automagically seen everywhere they should be, presumably because the ReSpec versions in use vary. There are appropriate places to use each of hyphens, n-dashes, and m-dashes, and the only way I have found to be sure they are seen where they should be is to explicitly put those characters in those places.

> 2/ White space around parentheses make no effort to find a middle ground on styling.

I've yet to hear a reasoned argument against my clearly stated reason (to wit, reader comprehension) for such whitespace; I've only heard personal taste, which IMNSHO should not be the deciding factor unless the WG as a whole (though perhaps not universally) decides that taste will govern all our output documents.

1. Below is a simple copy-paste from the [PR-Preview](https://pr-preview.s3.amazonaws.com/TallTed/sparql-update/pull/19.html). I see nothing problematic about this (though I do wonder whether some `&nbsp;` should be added before each closing brace or paren, and possibly following each opening brace or paren):
   > Note that, in the following, whenever we write Dataset-UNION ( X ) where X = { DS1, DS2, ..., DSn } is a set of datasets, we understand this as a shorthand for Dataset-UNION ( DS1, Dataset-UNION ( DS2, ..., Dataset-UNION ( DSn, {} ) ... ) ).

2. For discussion's sake, here's the same paragraph with the addition of backticks around what I think should actually be `<code>` styled in the document. I think this is even better than the above:
   > Note that, in the following, whenever we write `Dataset-UNION ( X )` where `X = { DS1, DS2, ..., DSn }` is a set of datasets, we understand this as a shorthand for `Dataset-UNION ( DS1, Dataset-UNION ( DS2, ..., Dataset-UNION ( DSn, {} ) ... ) )`.

3. In contrast, below is the same paragraph, without either the `<code>` styling or the extra whitespace. I find it harder to read, and see nothing that makes the "nested forms … look nested":
   > Note that, in the following, whenever we write Dataset-UNION( X ) where X = {DS1,DS2,... ,DSn} is a set of datasets, we understand this as a shorthand for Dataset-UNION(DS1, Dataset-UNION(DS2, ... , Dataset-UNION(DSn,{})...)).

4. Adding the `<code>` styling and still leaving out my whitespace doesn't help much, if at all:
   > Note that, in the following, whenever we write `Dataset-UNION( X )` where `X = {DS1,DS2,... ,DSn}` is a set of datasets, we understand this as a shorthand for `Dataset-UNION(DS1, Dataset-UNION(DS2, ... , Dataset-UNION(DSn,{})...))`.

I would be interested to hear other opinions on the above options, which I've numbered for easier reference.

**Comment by TallTed** [comment]
@gkellogg — These are indeed large documents, and the harder they are to understand, the less likely they are to be followed as intended and written. Because it will increase comprehension and compliance, I am happy to do a lot if not all of the initial round of work described above — as exemplified by this PR, which I would prefer to merge as a first step to be followed by other PRs, as opposed to continuing work *on* this PR, which will increasing accrue merge conflicts as other work is done.

**Comment by afs** [comment]
It is the reformatting of parentheses that concerns me the most.

Better to have consistent style - here, probably, redo the text as multiple lines and a definition box.

There is no point changing the parentheses in _every_ function, _every_ pattern evaluation definition, and more, in SPARQL.

**Comment by TallTed** [comment]
> It is the reformatting of parentheses that concerns me the most.
>
> Better to have consistent style - here, probably, redo the text as multiple lines and a definition box.

Possibly. Though I still think the extra whitespace makes things easier to read and understand, and I would advocate for that in your suggested "consistent style".

> There is no point changing the parentheses in _every_ function, _every_ pattern evaluation definition, and more, in SPARQL.

I disagree that there is _no_ point. As you say, "consistent style" is good.

**Comment by afs** [comment]
> I disagree that there is no point. As you say, "consistent style" is good.

The text in other important documents is consistent in style so use it - it was a lot of work to get it that way.

Is it perfect? Unlikely; but jumping on one mistake (in unrelated surrounding text in a PR for a single link fix for FPWD) and proposing the most change possible is not productive.

**Comment by TallTed** [comment]
> jumping on one mistake (in unrelated surrounding text in a PR for a single link fix for FPWD) and proposing the most change possible is not productive

I believe I've left that "PR for a single link fix" behind, some time ago.

I don't think this needs to be merged for FPWD.  But I also don't think that the current mostly consistent style is anywhere near optimal, and don't understand your objection to improving it.

**Comment by afs** [comment]
> I don't think this needs to be merged for FPWD.

See telecon discussion.

> is anywhere near optimal

Since you reject any comprise, let's not go round that again.

Use the same style as SPARQL Query.

This churn is blocking work on technical errata and CSS fixes.

**Comment by TallTed** [comment]
I heard nothing in the telecon that said "merge every outstanding PR before FPWD".

You have not offered any compromise; you've said "this is the way I'm doing it" and I continue to wait for a reasoned argument for that way that counters my arguments for the way of this PR.

I do not consider inertia nor personal taste to be a reasoned basis.

**Comment by afs** [comment]
Your argument was "I strongly believe (based on experience, which is unfortunately not citable research)"

> You have not offered any compromise;

- 1/ Do it properly with multiline formal definition.
- 2/ Do a minimal fix for now, remove the space for ` )`.

**Comment by TallTed** [comment]
> * Do it properly with multiline formal definition (and common programming, math typeset).
> * Common programming style (no spaces before or after `(`, none before `)`
> * A minimal fix for now, remove the space for ` )`.

I might be able to accept the first bullet, but your _"common programming" == no spaces around `(` nor `)` nor other punctuation_ does not work for me, partly because I do not like to encourage the programmer priesthood, and partly because this is not code meant to be compiled nor interpreted; it is educational material.

Programmers acting on this spec are welcome to leave out the syntactically *optional* spaces and other syntactically *optional* whitespace to shrink the size of their codebase.

Including the syntactically *optional* spaces and other syntactically *optional* whitespace in this _documentation_ will help increase comprehension by novices and non-programmers, and should not diminish comprehension by experts and/or programmers.

-----

I've addressed the merge conflicts to date. There remains a big chunk that was a conflict, being lines 1768-1836, which appears likely to have been moved to another document. If that could be confirmed, I will delete those lines from this PR.

**Comment by afs** [comment]
> lines 1768-1836, which appears likely to have been moved to another document

Worrying - what does the git log say?

**Comment by TallTed** [comment]
> Worrying - what does the git log say?

`<section id="sec-bibliography">` ... `</section>` appears to have been [removed from `spec/index.html`](https://github.com/w3c/sparql-update/pull/24/files) by #24

So, I'm dropping the same section, on that basis.

**Comment by afs** [comment]
I have no objection to this being merged.

**Review by domel — APPROVED** [review]

**Pull request merged**